### PR TITLE
Makefile to help with build and test commands

### PR DESCRIPTION
**Pull request by nader-ziada**
# Changes

- add a Makefile with build, test and lint commands
- running the lint locally helps avoid github actions failures

/kind enhancement

**Release Note**

```release-note
Add initial Makefile
```

### CODE REVIEW

File: hack/install-tools.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+echo '::group::🐶 Installing reviewdog ... https://github.com/reviewdog/reviewdog'
+curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b "${TOOLS_BIN_DIR}" 2>&1
+echo '::endgroup::'
+echo '::group:: Installing misspell ... https://github.com/client9/misspell'
+curl -sfL https://raw.githubusercontent.com/client9/misspell/master/install-misspell.sh | sh -s -- -b "${TOOLS_BIN_DIR}" 2>&1
+echo '::endgroup::'
+echo '::group:: Installing woke ... https://github.com/get-woke/woke'
+curl -sfL https://raw.githubusercontent.com/get-woke/woke/main/install.sh | sh -s -- -b "${TOOLS_BIN_DIR}" "${WOKE_VERSION}" 2>&1
+echo '::endgroup::'
```
Comment (zhanggbj) [inline hack/install-tools.sh:33]: Hi @nader-ziada 
Looks like `golangci-lint` installation is missing.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	hack/install-tools.sh
+
+## --------------------------------------
+## Lint
+## --------------------------------------
+
+
+.PHONY: go-lint
+go-lint:
+	golangci-lint -v run
```
Comment (zhanggbj) [inline Makefile:57]: It looks like `golangci-lint` is not installed by `make tools`.

Comment (nader-ziada) [reply Makefile]: added

File: hack/install-tools.sh (hunk 1 above)
Comment (nader-ziada) [reply hack/install-tools.sh]: added to install

**Review by zhanggbj — APPROVED** [review]
LGTM

**Pull request merged**